pandas/core/groupby/groupby.py

```diff
@@ -2735,6 +2746,10 @@ def _get_cythonized_result(
                 mask = isna(values).view(np.uint8)
                 func = partial(func, mask)
 
+            if how == "group_any_all" and isinstance(values, BaseMaskedArray):
+                post_processing = partial(post_processing, result_is_nullable=True)
+                func = partial(func, use_kleene_logic=True)
```

Comment:
It seems preferable to me to not special case here. Can use partial on post_processing in the caller. For func, would we instead use set `any, all, any_kleene, all_kleene` for val_test instead of an additional parameter?

Reply:
Thanks for the great suggestions! 

The problem I was running into is that the masked argument in `post_processing` and whether or not to use Kleene logic are not invariant in the loop in `_get_cythonized_result`. The caller doesn't know the nullable status because it might be possible that the first column is nullable, but the second column is not.

Definitely agree on trying to avoid special-casing, changed to provide a more general mechanism for nullable-dependent processing like this. While it would only be used for this case now, should make it easier to fix other nullable cases, such as std() or quantile() returning a float instead of Float.

Your suggestion for `any, all, any_kleene, all_kleene` instead of using a new parameter would be a neat way to avoid adding an additional parameter. However, couldn't figure out a clean way to do this without special casing any/all or adding more invasive mask-specific dispatch (any suggestions on this welcome!)